### ISSUE TITLE
react-is: 16.12.0 isMemo is returning wrong value

### ISSUE THREAD

**Issue by changran52m** [facebook/react#17457]
Bug in react-is: **16.12.0**
https://codesandbox.io/s/determined-lehmann-hl43r
![Screen Shot 2019-11-25 at 10 34 22 PM](https://user-images.githubusercontent.com/2319918/69605040-f029aa80-0fd3-11ea-9067-beb882662338.png)

This issue does not exist in ### 16.11.0
https://codesandbox.io/s/nifty-paper-8cqil

**Comment by milesj** [comment]
This changed in v16.12. https://github.com/facebook/react/pull/17278

**Comment by changran52m** [comment]
> This changed in v16.12: #17278

@milesj Thanks for your quick reply.
However, I am still be able to reproduce it here. Please correct me if I am wrong.
![Screen Shot 2019-11-25 at 10 52 47 PM](https://user-images.githubusercontent.com/2319918/69605990-7ba43b00-0fd6-11ea-84b5-9abbded02d3f.png)

https://codesandbox.io/s/determined-lehmann-hl43r
![Screen Shot 2019-11-25 at 10 34 22 PM](https://user-images.githubusercontent.com/2319918/69606012-878ffd00-0fd6-11ea-8129-5938b56fc9a5.png)

```
import React, { memo } from "react";

import { isMemo } from "react-is";

const Test = memo(() => {
  return <div>test</div>;
});

console.log("Test", isMemo(Test)); // false
```

**Comment by changran52m** [comment]
I think I get it now. It was a bug and get fixed in 16.12. But seems there are so many packages depending on the buggy function.

I will try some work around in our side and open issue in material UI and enzyme repo then.

**Comment by bvaughn** [comment]
As the linked PR #17278 mentions:
```js
const MemoizedComponent = React.memo(MyComponent);

// MemoizedComponent is a valid element TYPE.
expect(ReactIs.isValidElementType(MemoizedComponent)).toBe(true);

// <MemoizedComponent /> is a valid element.
expect(ReactIs.typeOf(<MemoizedComponent />)).toBe(ReactIs.Memo);
expect(ReactIs.isMemo(<MemoizedComponent />)).toBe(true);

// MemoizedComponent is NOT an element!
// (This used to return a false positive.)
expect(ReactIs.isMemo(MemoizedComponent)).toBe(false);
```

The previous behavior was broken. It's unfortunate that the bug fix caused some existing use cases to "break" but I think it was still the right change to make.

Sorry for the inconvenience!
